.github/workflows/docker-publish.yaml

```diff
@@ -37,6 +37,13 @@ jobs:
         run: echo "::set-output name=tag::$(echo ${GITHUB_REF##*/} | sed s/^v//g)"
         id: version
 
+      - name: Extract schema migration version
+        run: |
+          cd ./pkg/ddl
+          echo "::set-output name=version::$(echo $(ls -dq *up.sql* | wc -l))"
```

Comment:
Also not sure why you would want to `ls` in there, which in turns needs the `-d` and `-q` flags.  Also IIUC this counts the number of `.up.sql` files and _assumes_ that it will be the number that we want, i.e. that the count is right and that there are no "helper" `.up.sql` files.

Also not a fan of changing the working directory and having to remember to change it back -- particularly when in this case you don't actually need to change it back.

I suggest one of two alternatives:

1. Add an option to `lakefs` to print its schema number.  Now you can run it as part of the Docker build and plug that directly into a Dockerfile [`LABEL` command](https://docs.docker.com/engine/reference/builder/#label).  Advantages: Clean (doesn't require building from the workflow, which really looks like an accident waiting to happen), can use a higher-level language than bash to retrieve the desired schema version, generally useful lakefs command, uses the same migration library logic to determine the schema version.  Disadvantages: Sunk cost, some people might prefer shell script to Dockerfile.  (This is probably my preferred option.)
2. Count the files correctly, in shell.  E.g. because `ls` sorts its output (indeed that's its chief use in a shell script), this should probably work, because it asks sed to print only the number as the start of the last line:
   ```sh
   ...=version::$( (cd ../../pkg/ddl/ && ls -d *.up.sql) | sed -nE '$s/^([0-9]+).*/\1/p' )
   ```
   You might even strip leading zeroes by using the pattern `/^0*([0-9]+).*/` in there.  Advantages: brevity, bash golf.  Disadvantages: bash performance art, different route than migration library for determining the desired migration version, only work when you run the publish workflow.
3. Mix the shell command from (2) with running it as part of the Dockerfile and using an internal label command.

Reply:
I prefer option 1 as it's harder to make mistakes with it. The cost of a mistake in any of the alternatives is an image that is kind of broken *forever*, so any option we choose should be hardened (unit-test, integration, other..) from day 1.

Reply:
Also prefer option 1. This label sounds like a feature of the image, and hence should be computed in the Dockerfile.

Reply:
I've done some searching through documentation, and I don't believe there's currently a way for me to set the LABEL programmatically from within the Dockerfile. The db-schema-version is currently only obtainable by using RUN. LABELS can only be set manually or by using ARG / ENV variables, neither of which can be set to a value obtained using RUN. ARGs can be set from outside the Dockerfile when calling `docker build --build-arg`, but we could easily set the label with `docker build --label` as well. If we're outside the docker image (in the github workflow), I don't believe we can call `lakefs` to obtain the latest db-schema-version. 

I found [this stackoverflow question](https://stackoverflow.com/questions/31647843/labelling-images-in-docker) that I believe applies to this problem.

It seems like the shell script might be the way to go about this, unless there's some way of calling the lakeFS function from inside the workflow.

Reply:
> I don't believe there's currently a way for me to set the LABEL programmatically from within the Dockerfile.

:-(  Oh well...